### PR TITLE
EEG example notebook

### PR DESCRIPTION
**Pull request by bnaul**
Initial merge of a backend library tutorial using EEG data from `datasets.andrzejak`.

Also contains a couple of minor backend changes to make handling the EEG data easier.

### CODE REVIEW

File: mltsp/celery_tasks.py [hunk 1]
```diff
@@ -40,8 +55,66 @@ def featurize_ts_file(ts_data_file_path, features_to_use, metadata={},
...
+    Returns
+    -------
+    tuple
+        Two-element tuple whose first element is the file name and
+        second element is a dictionary of features.
+
+    """
+    try:
+        pickle.dumps(custom_functions)
+        if custom_functions:
```
Comment (stefanv) [inline mltsp/celery_tasks.py]: What is happening here?

File: mltsp/featurize.py [hunk 2]
```diff
@@ -20,26 +22,40 @@ def write_features_to_disk(featureset, featureset_id):
...
+    featureset = ft.assemble_featureset([], targets, meta_features)
     write_features_to_disk(featureset, featureset_id)
     return featureset
 
 
-def featurize_task_params_list(ts_paths, features_to_use, metadata=None,
+def featurize_data_task_params(times, values, errors, labels,
+                               features_to_use, meta_features=None,
+                               custom_script_path=None, custom_functions=None):
+    """Create list of tuples containing params for `featurize_data_task`."""
```
Comment (stefanv) [inline mltsp/featurize.py]: No parameter description in docstring?

File: mltsp/featurize.py [hunk 3]
```diff
@@ -20,26 +22,40 @@ def write_features_to_disk(featureset, featureset_id):
...
+            label = int(label)
+        params_list.append((t, m, e, label, features_to_use, meta_feature_dict,
+                            custom_script_path, custom_functions))
+    return params_list
+
+
+def featurize_file_task_params(ts_paths, features_to_use, meta_features=None,
                                custom_script_path=None):
-    """Create list of tuples containing params for `featurize_celery_task`."""
+    """Create list of tuples containing params for `featurize_file_task`."""
```
Comment (stefanv) [inline mltsp/featurize.py]: No parameter description in docstring

**Comment by stefanv** [comment]
Looks good overall; let's go over the patch later today, then you can explain some subtleties to me.

Also, do we want to store notebooks with/without output included?

File: mltsp/celery_tasks.py (hunk 1 above)
Comment (bnaul) [inline mltsp/celery_tasks.py]: This is just to catch the eventual pickle/celery failure in the case of unpickleable `custom_functions` being passed; I thought the exception text was good enough but I can add a comment if that'd be clearer.

File: mltsp/featurize.py (hunk 2 above)
Comment (bnaul) [inline mltsp/featurize.py]: Could I just say "refer to `featurize_time_series`"? The arguments are the same but the descriptions are long/complicated. Same with the function below.

**Comment by bnaul** [comment]
I'd vote for with! Don't see the downside.

**Comment by stefanv** [comment]
The downside is a big binary blob that has to be stored and messes up diffs.

File: Makefile [hunk 4]
```diff
@@ -5,7 +5,7 @@ PY_VER = $(shell python -c 'import sys; print(sys.version_info.major)')
 
 celery:
 	@if [[ -z `ps ax | grep -v grep | grep -v make | grep celery_tasks` ]]; then \
-		PYTHONPATH="./mltsp" celery worker -A celery_tasks -l info >>/tmp/celery.log 2>&1 & \
+		C_FORCE_ROOT=1 PYTHONPATH="./mltsp" celery worker -A celery_tasks -l info >>/tmp/celery.log 2>&1 & \
```
Comment (stefanv) [inline Makefile]: This should not be the default.

File: mltsp/celery_tasks.py (hunk 1 above)
Comment (stefanv) [inline mltsp/celery_tasks.py]: I meant the `all(f.__module__ != '__main__')` stuff.

File: mltsp/featurize.py (hunk 2 above)
Comment (stefanv) [inline mltsp/featurize.py]: Yup, that's perfectly fine.

File: mltsp/featurize.py [hunk 5]
```diff
@@ -241,20 +265,36 @@ def featurize_time_series(times, values, errors=None, features_to_use=[],
...
-                                          custom_functions=custom_functions)
-        feature_dicts.append(features)
+
+    if use_celery and celery_available():
+        params_list = featurize_data_task_params(times, values, errors, labels,
+                                                 features_to_use,
+                                                 meta_features,
+                                                 custom_script_path,
+                                                 custom_functions)
+        res = featurize_data_task.chunks(params_list, cfg.N_CORES).delay()
```
Comment (bnaul) [inline mltsp/featurize.py]: `celery_result =` ?

**Comment by bnaul** [comment]
Build looking good; I went ahead and fixed the test errors in a way that will work with or without `netcdf4`. Everything else we talked about is in there!

**Pull request merged**